lighthouse-core/test/report/html/renderer/report-renderer-test.js

```diff
@@ -201,6 +201,51 @@ describe('ReportRenderer', () => {
       assert.ok(/\dx/.test(descriptionsTxt), 'should have added CPU');
       assert.ok(descriptionsTxt.includes(sampleResults.userAgent), 'user agent populated');
     });
+
+    describe('axe-core', () => {
+      let axe;
+
+      beforeAll(() =>{
+        // Needed by axe-core
+        // https://github.com/dequelabs/axe-core/blob/581c441c/doc/examples/jsdom/test/a11y.js#L24
+        global.window = global.self;
+        global.Node = global.self.Node;
+        global.Element = global.self.Element;
+
+        axe = require('axe-core');
+      });
+
+      afterAll(() => {
+        global.window = undefined;
+        global.Node = undefined;
+        global.Element = undefined;
+      });
+
+      it('renders without axe violations', (done) => {
+        const output = renderer.renderReport(sampleResults, renderer._dom._document.body);
+
+        const config = {
+          rules: {
+            // Reports may have duplicate ids
+            // https://github.com/GoogleChrome/lighthouse/pull/5058#discussion_r184791923
+            'duplicate-id': {enabled: false},
+            // The following rules are disable for axe-core + jsdom compatibility
+            // https://github.com/dequelabs/axe-core/tree/b573b1c1/doc/examples/jest_react#to-run-the-example
+            'color-contrast': {enabled: false},
+            'link-in-text-block': {enabled: false},
+          },
+        };
+
+        axe.run(output, config, (error, {violations}) => {
+          assert.ifError(error);
+          assert.ok(violations.length === 0, JSON.stringify(violations, null, 1));
```

Comment:
`expect(violations).toHaveLength(0)` (this also shows you the entire array if it fails).

Reply:
The output is not formatted :( I will stick with the current JSON.stringify unless the formatting is not an issue.
![axe-core-failure-message-2](https://user-images.githubusercontent.com/309310/61575002-524d2e80-aa7b-11e9-9d5e-2a0d33a088ef.PNG)

Reply:
FWIW I'd also prefer `expect(violations).toHaveLength(0)`. It makes the intention super clear compared to the generic assert. The debugging story of the JSON is slightly worse, but the debugging story of the JSON is pretty bad no matter what because it's so huge, and it's still the improvement to initial readability of the failure and code is a win to me